core/src/system.rs

```diff
@@ -259,6 +263,54 @@ impl System {
                 }
             }
 
+            CoreRunOutcome::ReservedPidInterfaceMessage {
+                pid,
+                message_id,
+                interface,
+                message,
+            } if interface == redshirt_pipeline_interface::ffi::INTERFACE => {
+                
+                match redshirt_pipeline_interface::ffi::PipelineMessage::decode(message) {
+                    Ok(redshirt_pipeline_interface::ffi::PipelineMessage { module, funcname, inputs }) => {
+                        let module = Module::from_bytes(&module).expect("module isn't proper wasm");
+
+                        let instance = wasmi::ModuleInstance::new(
+                                module.as_ref(),
+                                &wasmi::ImportsBuilder::default()
+                            )
+                            .expect("failed to instantiate wasm module")
+                            .assert_no_start();
+
+                        let result = match instance.invoke_export(
```

Comment:
Redshirt uses a scheduler to run programs. Here, we do not, but I need to figure out if I can also use the scheduler for calling wasm functions.